core/src/main/java/apoc/trigger/TriggerHandler.java

```diff
@@ -203,8 +204,17 @@ public Void beforeCommit(TransactionData txData, Transaction transaction, GraphD
 
     @Override
     public void afterCommit(TransactionData txData, Void state, GraphDatabaseService databaseService) {
+        // if `txData.metaData()` is equal to TRIGGER_META, it means that the transaction
+        // if it is equal, it means that the transaction comes from another TriggerHandler transaction,
+        // therefore the execution must be blocked to prevent a deadlock due to cascading transactions
```

Comment:
```suggestion
        // if `txData.metaData()` is equal to TRIGGER_META, it means that the transaction
        // comes from another TriggerHandler transaction,
        // therefore the execution must be blocked to prevent a deadlock due to cascading transactions
```